### PR TITLE
changelog: fix disabled_algorithms suggestion for legacy servers

### PR DESCRIPTION
**Pull request by perrinjerome**
Fixes https://github.com/paramiko/paramiko/issues/2090

### CODE REVIEW

File: sites/www/changelog.rst [hunk 1]
```diff
@@ -246,7 +246,7 @@ Changelog
     required if your target systems do not support either RSA2 or the
     ``server-sig-algs`` protocol extension.
 
-    Specifically, you need to specify ``disabled_algorithms={'keys':
+    Specifically, you need to specify ``disabled_algorithms={'pubkeys':
```
Comment (bskinn) [inline sites/www/changelog.rst:249]: Is `pubkeys` the correct key to use in `disabled_algorithms` in *all* cases? Or is the behavior more complex than that? (Honest question, I'm not deeply familiar.)

Comment (perrinjerome) [reply sites/www/changelog.rst:249]: It's always hard to tell about *all* cases so the short answer is "I am not sure", but many things make me believe this was a typo.

This changelog entry has a more detailed explanation below and here it is mentioned to use `pubkeys`:

https://github.com/paramiko/paramiko/blob/4bf5d30e0dd6eb7ebbecc03445c3b77f61ad761d/sites/www/changelog.rst?plain=1#L275-L276

Using `pubkeys` was the solution for  https://github.com/paramiko/paramiko/issues/2090#issuecomment-1192829965 and also for me when trying to connect to a "XFB.Gateway" server.

Also, this is documented as a workardound for servers not supporting the [`server-sig-algs`]( https://www.rfc-editor.org/rfc/rfc8308#section-3.1) extension and this is an extension for public key algorithms.

`keys` was added in https://github.com/paramiko/paramiko/commit/5f222495b5a62f3a1c465292bcace15888f40515 maybe @bitprophet can confirm ? Thanks !

**Comment by bitprophet** [comment]
Thanks for posting! But nah, the original changelog should have been correct, the issue it was correcting for was _host keys_ related which, for legacy reasons, is what `keys` maps to here. `pubkeys` is specifically about user authentication, which as you found, is listed elsewhere in the changelog.

It's a little buried / requires the user to briefly check the code to see which attribute they're trying to mess with (and that's also where the confusion about keys vs pubkeys arises), but `disabled_algorithms`' format is doc'd here: https://docs.paramiko.org/en/stable/api/transport.html#paramiko.transport.Transport.__init__

This will all get _much_ better soon once I have a newer, cleaner auth system available for opt-in (being tracked under #387 for now).

**Comment by Kache** [comment]
Then perhaps the docs should have a short list of the different keys users would need to address the ["slightly backwards incompatible" change](https://www.paramiko.org/changelog.html#2.9.0)?

Perhaps like:

 * `keys` - host keys
 * `pubkeys` - user auth
 * Others, plus link to source where `Keys should match the last word in the class’ builtin algorithm tuple attributes` is defined

**Pull request closed without merging**